control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go

```diff
@@ -71,9 +74,37 @@ func ReconcileNetworkOperator(network *operatorv1.Network, networkType hyperv1.N
 
 	// Setting the management state is required in order to create
 	// this object. We need to create this object before the cno starts
-	// because mutating many of the values (like vxlanport) is not premitted
+	// because mutating many of the values (like vxlanport) is not permitted
 	// after the cno reconciles this operator CR
 	if network.Spec.ManagementState == "" {
 		network.Spec.ManagementState = "Managed"
 	}
 }
+
+func DetectSuboptimalMTU(ctx context.Context, mgmtClient client.Client,
+	guestNetworkOperator *operatorv1.Network, hcp *hyperv1.HostedControlPlane) error {
+	const recommendedMinMTU = uint32(9000)
+	const ovnkOverhead = 200
+	if hcp.Spec.Platform.Type == hyperv1.KubevirtPlatform &&
+		hcp.Spec.Networking.NetworkType == hyperv1.OVNKubernetes &&
+		guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig != nil &&
+		guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig.MTU != nil &&
+		*guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig.MTU < recommendedMinMTU-ovnkOverhead {
+		meta.SetStatusCondition(&hcp.Status.Conditions, metav1.Condition{
+			Type:               string(hyperv1.SubOptimalMTUDetected),
+			Status:             metav1.ConditionTrue,
+			Reason:             hyperv1.KubeVirtSuboptimalMTUReason,
+			ObservedGeneration: hcp.Generation,
+			Message: fmt.Sprintf("A suboptimal MTU size of %d bytes has been detected. "+
+				"Due to performance, we recommend setting an MTU of %d bytes or greater for the "+
+				"infra cluster that hosts the KubeVirt VirtualMachines. When smaller MTUs are used, the cluster will "+
+				"still operate, but network performance will be degraded due to fragmentation of the double "+
+				"encapsulated ovn-k.",
+				*guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig.MTU+ovnkOverhead, recommendedMinMTU),
+		})
+		if err := mgmtClient.Status().Update(ctx, hcp); err != nil {
+			return fmt.Errorf("failed to set suboptimal MTU condition on HCP: %w", err)
+		}
```

Comment:
do we need a semantic equals here to ensure we only update when something changes?

Reply:
Done.
HCP is not being updated if there is no change.